File: tags/v0.2.0/ruby/demoribclient.rb
require 'ribclient'

r = Ribclient


filename = "demoribclient.rib"
display = "demoribclient.tif"
eocdisplay = "+eoc.tif"

r.Option("rib",{"compression"=>["gzip"]})

r.Begin(filename)

r.Option("rib",{"format"=>["binary"]})

r.FrameBegin(1)
r.Format(640,480,1.0)

r.Projection(Ribclient::PERSPECTIVE,{Ribclient::FOV=>[27]})

r.Display(display,Ribclient::FRAMEBUFFER,Ribclient::RGBA,nil)

r.WorldBegin

lhd = r.LightSource(Ribclient::DISTANTLIGHT,{Ribclient::INTENSITY=>[2.0]})

r.AttributeBegin
r.Translate(0,0,10)
r.Sphere(1,-1,1,360,nil)
r.AttributeEnd

r.WorldEnd

r.FrameEnd

r.End

File: python/demoribclient.py
# Copyright Procedural Insight http://www.proceduralinsight.com
# Author : Nicholas Yue

import ribclient

ri = ribclient

ri.Option('rib',{'compression':['gzip']})

ri.Begin('demoribclient.rib')

ri.Option('rib',{'format':['binary']})

ri.FrameBegin(1)
ri.Format(640,480,1.0)

ri.Projection(ri.PERSPECTIVE,{ri.FOV:[27.0]})

ri.Display('demoribclient.tif',ri.FRAMEBUFFER,ri.RGBA,None)

ri.WorldBegin()
    
plh = ri.LightSource(ri.DISTANTLIGHT,{ri.INTENSITY:[2],ri.LIGHTCOLOR:[1,1,1]})

ri.AttributeBegin()
ri.Translate(0,0,10)
ri.Sphere(1,-1,1,360,None)
ri.AttributeEnd()

ri.WorldEnd()

ri.FrameEnd()
ri.End()
